File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int i=0; i<nums.length; i++){
            if (!map.containsKey(nums[i])){
                map.put(nums[i], 0);
            }
            else {return true;}
        }
        return false;
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-0.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        if (nums.size() == 0) return {};
        if (nums.size() == 1) return {nums[0], nums[0]};
        vector<int> ans(2 * nums.size());
        for (int i = 0; i < nums.size(); i++) {
            ans[i] = nums[i];
            ans[i + nums.size()] = nums[i];
        }
        return ans;
    }
};